warehouse/query-core/src/main/java/datawave/query/jexl/visitors/TreeFlatteningRebuilder.java

```diff
@@ -125,6 +142,15 @@ else if (hasChildren && node == parentStack.peek()) {
             // if we still have nodes to evaluate
             if (!postOrderStack.isEmpty()) {
                 
+                if (pruned) {
+                    if (!parentStack.isEmpty() && parentStack.peek() == postOrderStack.peek()) {
+                        childrenStack.peek().add(newNode);
+                    } else if (!postOrderStack.isEmpty()) {
+                        parentStack.push(postOrderStack.peek());
+                        childrenStack.push(Lists.newArrayList(newNode));
+                    }
+                } else
```

Comment:
Dangling else...

Reply:
Removing this else will cause a bunch of test failures. The execution flow is expecting to loop back around before making any changes to the nodes.